tests/func/test_remote_dir.py

```diff
@@ -38,7 +51,54 @@ def remote():
         yield remote
 
 
-def test_isdir(remote):
+@pytest.fixture
+def remote_gs():
+    """Returns a RemoteGS connected to a bucket with the following structure:
+        bucket
+        ├── data
+        │  ├── alice
+        │  ├── alpha
+        │  └── subdir
+        │     ├── 1
+        │     ├── 2
+        │     └── 3
+        ├── empty_dir
+        ├── empty_file
+        └── foo
+    """
+    prefix = str(uuid.uuid4())
+    REMOTE_URL = "gs://dvc-test/" + prefix
+    remote = RemoteGS(None, {"url": REMOTE_URL})
+    teardowns = []
+
+    def put_object(file, content):
+        create_object_gs(remote.gs, "dvc-test", prefix + "/" + file, content)
+        teardowns.append(lambda: remote.remove(remote.path_info / file))
+
+    put_object("empty_dir/", "")
+    put_object("empty_file", "")
+    put_object("foo", "foo")
+    put_object("data/alice", "alice")
+    put_object("data/alpha", "alpha")
+    put_object("data/subdir/1", "1")
+    put_object("data/subdir/2", "2")
+    put_object("data/subdir/3", "3")
```

Comment:
This duplicates the structure. And since the tests are the same these two need to stay in sync, let's refactor it somehow to remote duplication.

Maybe the following - remote desc classes from `test_api.py` - will help.